cmd/exo/cmd/request.go

```diff
@@ -21,26 +21,31 @@ type taskStatus struct {
 	jobStatus egoscale.JobStatusType
 }
 
-// asyncTasks message variable must have same size with cmds
-func asyncTasks(tasks []task) ([]interface{}, []error) {
+type taskResponse struct {
+	resp interface{}
+	error
+}
+
+func asyncTasks(tasks []task) []taskResponse {
 
 	//init results
-	e := make([]error, 0, len(tasks))
-	errors := &e
-	r := make([]interface{}, len(tasks))
-	responses := &r
+	responses := make([]taskResponse, len(tasks))
 
 	//create task Progress
 	taskBars := make([]*mpb.Bar, len(tasks))
 	maximum := 10
-	var wg sync.WaitGroup
-	p := mpb.New(mpb.WithWaitGroup(&wg), mpb.WithContext(gContext), mpb.WithWidth(40))
-	wg.Add(len(tasks))
+	var taskWG sync.WaitGroup
+	p := mpb.New(mpb.WithWaitGroup(&taskWG), mpb.WithContext(gContext), mpb.WithWidth(40))
+	taskWG.Add(len(tasks))
+
+	var workerWG sync.WaitGroup
+	workerWG.Add(len(tasks))
+	workerSem := make(chan int, len(tasks))
```

Comment:
if the semaphore is as big as the list of tasks, you're not doing anything different.

Reply:
yes sure so if i want to do all tasks at the same time i can remove semephore !
or for example i can run it 4 by 4 with semaphore for example ?

Reply:
marco says something like 10 or 20.

Reply:
also no magical numbers, so I'd create a proper const for that too.

Reply:
yep yep thx

Reply:
20 LGTM